src/orange/detail/UnitInserter.cc

```diff
@@ -186,6 +192,10 @@ SimpleUnitId UnitInserter::operator()(const UnitInput& inp)
     }
 
     // Save unit scalars
+    if (inp.volumes.back().zorder == 1)
+    {
+        unit.background = VolumeId(inp.volumes.size() - 1);
+    }
```

Comment:
Is there any risk that a file would have a zorder == 1 volume that's _not_ the last item?

Reply:
In the current SCALE ORANGE implementation no. Since we always search volumes in decreasing zorder, 1 is the lowest valid value. If I'd been a little more foresighted I would've used strings for this (and the flags) to reduce the chance of errors cropping up across the translation layer. Hopefully soon though we can delete the translation layer.